dipy/align/mattes.pyx

```diff
@@ -1219,10 +1219,10 @@ cdef double _compute_mattes_mi(double[:, :] joint,
 
             if mi_gradient is not None:
                 for k in range(n):
-                    mi_gradient[k] -= joint_gradient[i, j, k] * factor
```

Comment:
In the original paper by Mattes, he uses the negative MI, and I coded everything like in the paper, but it may be misleading to the users if we return the negative MI. The change of sign (to do minimization instead of maximization) will be made at the last moment, after we call this from the optimizer.